Bottom: f9161cbeea78baaece6165620e980e819e04d8a1
Top:    035ff3476d4e67a8deef315f2ab917127bb7cc7d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-06 22:28:09 +0100

hid/gtk: Consolidate the mechanisms used for holding off scrollbar updates

Don't duplicate mechanisms - it is confusing!


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 87a391b..40e6eef 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -58,12 +58,6 @@ ghid_port_ranges_changed (void)
 {
   GtkAdjustment *h_adj, *v_adj;
 
-  if (ghidgui->combine_adjustments)
-    {
-      ghidgui->combine_adjustments = FALSE;
-      return;
-    }
-
   h_adj = gtk_range_get_adjustment (GTK_RANGE (ghidgui->h_range));
   v_adj = gtk_range_get_adjustment (GTK_RANGE (ghidgui->v_range));
   gport->view_x0 = h_adj->value;
@@ -104,12 +98,15 @@ ghid_port_ranges_pan (gdouble x, gdouble y, gboolean relative)
   if (y1 > v_adj->upper - v_adj->page_size)
     y1 = v_adj->upper - v_adj->page_size;
 
-  if (x0 != x1 && y0 != y1)
-    ghidgui->combine_adjustments = TRUE;
-  if (x0 != x1)
-    gtk_range_set_value (GTK_RANGE (ghidgui->h_range), x1);
-  if (y0 != y1)
-    gtk_range_set_value (GTK_RANGE (ghidgui->v_range), y1);
+  ghid_screen_update ();
+
+  ghidgui->adjustment_changed_holdoff = TRUE;
+  gtk_range_set_value (GTK_RANGE (ghidgui->h_range), x1);
+  gtk_range_set_value (GTK_RANGE (ghidgui->v_range), y1);
+  ghidgui->adjustment_changed_holdoff = FALSE;
+
+  if ((x0 != x1) || (y0 != y1))
+    ghid_port_ranges_changed();
 
   ghid_note_event_location (NULL);
   return ((x0 != x1) || (y0 != y1));
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index 651a746..5a51358 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -126,7 +126,7 @@ typedef struct
   gboolean adjustment_changed_holdoff,
     toggle_holdoff,
     command_entry_status_line_active,
-    auto_pan_on, in_popup, combine_adjustments;
+    auto_pan_on, in_popup;
 
   gboolean config_modified,
     small_label_markup,
